### PR TITLE
chore: Updates mongodbatlas_advanced_cluster tests to expect temporary SERVICE_UNAVAILABLE error when migrating from old to new schema

### PR DESCRIPTION
**Pull request by maastha**
## Description

Updates mongodbatlas_advanced_cluster tests to expect temporary SERVICE_UNAVAILABLE error when migrating from old to new schema

Link to any related issue(s):[CLOUDP-269298](https://jira.mongodb.org/browse/CLOUDP-269298)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Further comments
Test runs with this change:

[TestMigAdvancedCluster_shardedMigrationFromOldToNewSchema](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/10523932274/job/29159669387):white_check_mark:
[TestMigAdvancedCluster_geoShardedMigrationFromOldToNewSchema](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/10523940528/job/29159694022):white_check_mark:

### CODE REVIEW

File: internal/service/networkpeering/resource_network_peering.go [hunk 1]
```diff
@@ -9,14 +9,16 @@ import (
 	"strings"
 	"time"
 
+	"go.mongodb.org/atlas-sdk/v20240805001/admin"
```
Comment (maastha) [inline internal/service/networkpeering/resource_network_peering.go:12]: ignore changes to network_peering in this PR, will remove

File: internal/service/advancedcluster/resource_advanced_cluster_migration_test.go [hunk 2]
```diff
@@ -104,6 +105,11 @@ func TestMigAdvancedCluster_shardedMigrationFromOldToNewSchema(t *testing.T) {
 				Config:            configShardedTransitionOldToNewSchema(orgID, projectName, clusterName, false),
 				Check:             checkShardedTransitionOldToNewSchema(false),
 			},
+			{
+				ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+				Config:                   configShardedTransitionOldToNewSchema(orgID, projectName, clusterName, true),
+				ExpectError:              regexp.MustCompile("SERVICE_UNAVAILABLE"),
```
Comment (AgustinBettati) [inline internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:111]: just to keep in mind, after service unavailable response migration is run asynchronously. We might have the risk following patch is done before migration is completed, so we should keep an eye on these tests and potentially add some delay before the 3rd step is run.

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: internal/service/advancedcluster/resource_advanced_cluster_migration_test.go (hunk 2 above)
Comment (maastha) [reply internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:111]: do you have an idea of how we can add a delay in the test steps? I couldn't find anything relevant on this

Comment (EspenAlbert) [reply internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:111]: @maastha 
I have done this before using a custom check.
[See this example](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2474/files#diff-2c1e29330aa05fe41eb92d73586162bda70abc84d747ec203eb159663d2e37bbR67)

Comment (maastha) [reply internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:111]: Thanks @EspenAlbert , @AgustinBettati does this work for you? lmk I'll update, afaik there's no built-in way to do this so I'm okay with the approach suggested by Espen.

Comment (AgustinBettati) [reply internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:111]: was imagining something along those lines 👍 , thanks Espen for sharing

**Pull request merged**